web/yo/app/scripts/controllers/reportgenerator.js

```diff
@@ -204,4 +204,264 @@ angular.module('oncokbApp')
+                } else {
+                    mainUtils.getOncoTreeMainTypes().then(function(result) {
+                        _.each(result.tumorTypes, function(items) {
+                            _.each(items, function(item) {
+                                subtypeMapping[item.code] = item.name;
+                            });
+                        });
+                        if ($scope.evidenceType === 'tumor_summary') {
+                            $http.get(OncoKB.config.publicApiLink + 'evidences/lookup?source=oncotree&evidenceTypes=TUMOR_TYPE_SUMMARY').then(function (response) {
+                                _.each(response.data, function (item) {
+                                    if (item.subtype) {
+                                        reviewedData.tumor_summary.push({
+                                            gene: item.gene.hugoSymbol,
+                                            mutation: getAlterations(item.alterations),
+                                            tumor_type: subtypeMapping[item.subtype],
+                                            tumor_summary: item.description
+                                        });
+                                    } else {
+                                        reviewedData.tumor_summary.push({
+                                            gene: item.gene.hugoSymbol,
+                                            mutation: getAlterations(item.alterations),
+                                            tumor_type: item.cancerType,
+                                            tumor_summary: item.description
+                                        });
+                                    }
+                                });
+                                $scope.reviewedData.tumor_summary = reviewedData.tumor_summary;
+                                finishLoadingReviewedData();
+                            });
+                        } else if ($scope.evidenceType === 'drugs') {
+                            $http.get(OncoKB.config.publicApiLink + 'evidences/lookup?source=oncotree&evidenceTypes=STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY%2C%20STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_RESISTANCE%2C%20INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY%2C%20INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_RESISTANCE').then(function (drugResponse) {
+                                loopArray(drugResponse.data, 0);
+                            });
+                        }
+                    });
+                } 
+            }
+            function getAlterations(alterations) {
+                var result = [];
+                _.each(alterations, function (item) {
+                    result.push(item.alteration);
+                });
+                return result.join();
+            }
+            function getCitations(data) {
+                var processedData = FindRegex.result(data);
+                var PMIDs = [];
```

Comment:
The purpose of this function is getting the citation for two different formats. But they are all string arrays. Why not just use one array? Order matters? If so, use unshift and push?

Reply:
We want to put all PMIDs together and all abstracts together.